### ISSUE TITLE
Jimp breaks Bluebird - PromisifyAll is not a function

### ISSUE THREAD

**Issue by ananaszjoe** [jimp-dev/jimp#674]
# Expected Behavior
Jimp should not break other modules

# Current Behavior
I use Bluebird and Raven like so:
```
global.Promise = require('bluebird')
const raven = global.Promise.promisifyAll(require('raven'))
```
if I require Jimp in my project like so,
`var Jimp = require('jimp')`
It breaks Bluebird's pormisifyAll function and I get the following error:
`TypeError: global.Promise.promisifyAll is not a function`

When I only remove `var Jimp = require('jimp')` from my poject, it runs without problem.

## Steps to Reproduce
1. Include and use Bluebird and Raven like so (app.js):
```
global.Promise = require('bluebird')
const raven = global.Promise.promisifyAll(require('raven'))
```
2. Include Jimp in the project (for me it's in a separate file for a controller):
`var Jimp = require('jimp')`

3. run `$ node app`

## Context

- Jimp Version: 0.6.0
- Operating System: Windows 10 Home version 1803 build 17134.407
- Node version: 8.12.0

## Failure Logs

```
> npm run start

> oa-api@0.1.0 start <redacted>
> node app

WARN: `increment` should not be used as a schema path name unless you have disabled versioning.
C:\<redacted>\app.js:12
const raven = global.Promise.promisifyAll(require('raven')) // Sentry.io
                             ^

TypeError: global.Promise.promisifyAll is not a function
    at Object.<anonymous> (C:\<redacted>\app.js:12:30)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! oa-api@0.1.0 start: `node app`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the oa-api@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\<redacted>\AppData\Roaming\npm-cache\_logs\2018-12-08T13_32_44_404Z-debug.log
```

**Comment by hipstersmoothie** [comment]
That's weird as hell. Will def look into this this week

**Comment by Jakovb** [comment]
Can confirm, any eta?

**Comment by oliversalzburg** [comment]
https://github.com/feathersjs/feathers/issues/510

Using `global` is not a great idea.

**Comment by oliversalzburg** [comment]
@hipstersmoothie I was suggesting that babel-polyfill clobbers `global.Promise`. I don't understand why they do that if a symbol exists at that location, but it'd be an issue with babel-polyfill in that case.

In general, I would recommend to not use `global` in user code. If you want bluebird, just `const Promise = require("bluebird");` everywhere where you want to use it.

That being said, the actual on-disk modules that are produced by jimp are a horror to debug and I don't think a babel-polyfill should be loaded by a dependency in a NodeJS project...

**Comment by oliversalzburg** [comment]
> What would you suggest doing instead of babel polyfill to support old versions of node?

Don't support old versions. If the consumer of the library wants to use it with an older version of Node, have _them_ include a polyfill where it makes sense for them.

> Also how are they a horror to debug? 

When I set a breakpoint on the `require("jimp")` call and step into the module code, I find it very hard to follow. Especially if your goal is to find out where something might register a symbol on `global`. But that might just be me. I just don't like debugging transpiled code. And I only got into this issue randomly. I'm not even affected by it. So I guess we can ignore my personal preference here.
